### ISSUE TITLE
Failed to find accessibility element with the label

### ISSUE THREAD

**Issue by xxi511** [kif-framework/KIF#987]
Hi, There is a similar closed issue #936 , but it seems no information about how to resolve.
Test on simulator is work fine, but failed on real device

Here is my log
seems the failed caused by setting `AccessibilityEnabled ` and `ApplicationAccessibilityEnabled `
```
[User Defaults] Failed to write value for key ApplicationAccessibilityEnabled in CFPrefsPlistSource<0x174110620> (Domain: com.apple.Accessibility, User: kCFPreferencesCurrentUser, ByHost: No, Container: (null)): setting preferences outside an application's container requires user-preference-write or file-write-data sandbox access, switching to read-only

[User Defaults] attempt to set 1 for key in AutomationEnabled in read-only (due to a previous failed write) preferences domain CFPrefsPlistSource<0x174110620> (Domain: com.apple.Accessibility, User: kCFPreferencesCurrentUser, ByHost: No, Container: (null))

[User Defaults] attempt to set 1 for key in ApplicationAccessibilityEnabled in read-only (due to a previous failed write) preferences domain CFPrefsPlistSource<0x174110620> (Domain: com.apple.Accessibility, User: kCFPreferencesCurrentUser, ByHost: No, Container: (null))

[User Defaults] attempt to set 1 for key in AccessibilityEnabled in read-only (due to a previous failed write) preferences domain CFPrefsPlistSource<0x174110620> (Domain: com.apple.Accessibility, User: kCFPreferencesCurrentUser, ByHost: No, Container: (null))

// 10 sec latter, throw exception
The step timed out after 10.00 seconds: Failed to find accessibility element with the label "THE LABEL"
(null)
```

**Environment**
Project language: Objective-C
Test language: swift 3
Xcode: 8.3.2 (8E2002)
Device: iPhone 6
iOS version: 10.3.1 (14E304)
Setting -> Developer -> UI AUTOMATION -> Enable UI Automation (ON)

Do I might miss some step? or need more information?

**Comment by mikelupo** [comment]
If you can put together a small foo project that can be shared and reproduces the error it would help us to help you.

**Comment by xxi511** [comment]
I tried to create a small project, but the new project is work fine...
It show the same log as I pasted above `[User Defaults] Failed to write` .... but the test is successful
it find element by accessibility label success

I know it hard, but can you give me some suggestion to try?
thanks

**Comment by justinseanmartin** [comment]
If you put a breakpoint in `-[KIFTestCase failWithException:stopTest:]`, you should be able to run `po [UIView printViewHierarchy]` when it is failing to inspect the app state. You could also set a breakpoint when the UIView you're attempting to find is created and check that its `accessibilityLabel` property is reflecting the value that you're searching for.

I'm not sure what you mean by "Setting -> Developer -> UI AUTOMATION -> Enable UI Automation (ON)". Could you send a screenshot? 

Either way, that shouldn't be necessary to run KIF tests, as it is built on the Unit Test framework. Please make sure that your test target is a Unit Test bundle.

Unfortunately, I don't personally use KIF for testing on physical devices. If it is something specific for getting accessibility enabled properly for physical devices that has changed, I don't have any insight.

**Comment by xxi511** [comment]
Sorry for reply late

it can test on real device now, but I don't know what I did
one of problem is accessibility label seems unavailable for `UITabBarButton` in real device, #457 

Setting -> Developer -> UI AUTOMATION -> Enable UI Automation (ON) is 
![image](https://user-images.githubusercontent.com/10164640/28919713-4b328fec-7881-11e7-8dd2-4655db600af0.png)
